"hack/update-bazel.sh" with several complaints about "too few values in struct initializer" on release 1.8

<!-- This form is for bug reports and feature requests ONLY! 

If you're looking for help check [Stack Overflow](https://stackoverflow.com/questions/tagged/kubernetes) and the [troubleshooting guide](https://kubernetes.io/docs/tasks/debug-application-cluster/troubleshooting/).
-->

**Is this a BUG REPORT or FEATURE REQUEST?**:
/kind bug
> Uncomment only one, leave it on its own line: 
>
> /kind bug
> /kind feature


**What happened**:
```
root@andy-dev:/code2/go/src/k8s.io/kubernetes# git checkout upstream/release-1.8
HEAD is now at 0c7e7ae... Merge pull request #60014 from cofyc/automated-cherry-pick-of-#59170-upstream-release-1.8
root@andy-dev:/code2/go/src/k8s.io/kubernetes# hack/update-bazel.sh
Cloning into '/tmp/tmp.PVJt57R744/go/src/github.com/kubernetes/repo-infra'...
remote: Counting objects: 848, done.
remote: Total 848 (delta 0), reused 0 (delta 0), pack-reused 848
Receiving objects: 100% (848/848), 261.63 KiB | 0 bytes/s, done.
Resolving deltas: 100% (385/385), done.
Checking connectivity... done.
Cloning into '/tmp/tmp.PVJt57R744/go/src/github.com/bazelbuild/rules_go'...
remote: Counting objects: 9265, done.
remote: Compressing objects: 100% (26/26), done.
remote: Total 9265 (delta 8), reused 11 (delta 2), pack-reused 9237
Receiving objects: 100% (9265/9265), 2.09 MiB | 0 bytes/s, done.
Resolving deltas: 100% (5983/5983), done.
Checking connectivity... done.
# github.com/bazelbuild/rules_go/go/tools/gazelle/merger
go/tools/gazelle/merger/fix.go:88:28: too few values in struct initializer
go/tools/gazelle/merger/merger.go:491:19: too few values in struct initializer
go/tools/gazelle/merger/merger.go:495:19: too few values in struct initializer
# github.com/bazelbuild/rules_go/go/tools/gazelle/rules
go/tools/gazelle/rules/generator.go:131:16: too few values in struct initializer
go/tools/gazelle/rules/sort_labels.go:30:16: too few values in struct initializer
!!! [0227 06:04:38] Call tree:
!!! [0227 06:04:38]  1: hack/update-bazel.sh:29 kube::util::go_install_from_commit(...)
!!! Error in /code2/go/src/k8s.io/kubernetes/hack/lib/util.sh:568
  Error in /code2/go/src/k8s.io/kubernetes/hack/lib/util.sh:568. '((i<3-1))' exited with status 2
Call stack:
  1: /code2/go/src/k8s.io/kubernetes/hack/lib/util.sh:568 kube::util::go_install_from_commit(...)
  2: hack/update-bazel.sh:29 main(...)
Exiting with status 1
!!! [0227 06:04:38] Call tree:
!!! [0227 06:04:38]  1: hack/update-bazel.sh:29 kube::util::go_install_from_commit(...)
!!! Error in /code2/go/src/k8s.io/kubernetes/hack/lib/util.sh:553
  Error in /code2/go/src/k8s.io/kubernetes/hack/lib/util.sh:553. '((i<3-1))' exited with status 1
Call stack:
  1: /code2/go/src/k8s.io/kubernetes/hack/lib/util.sh:553 kube::util::go_install_from_commit(...)
  2: hack/update-bazel.sh:29 main(...)
Exiting with status 1
```

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:
```
# git checkout upstream/release-1.8
# hack/update-bazel.sh
```

**Anything else we need to know?**:
"hack/update-bazel.sh" on v1.7, v1.9, v1.10 all works, only on v1.8 returns error.

**Environment**:
- Kubernetes version (use `kubectl version`): v1.8
- Cloud provider or hardware configuration:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

--- a/hack/lib/util.sh
+++ b/hack/lib/util.sh
@@ -550,9 +550,11 @@ kube::util::ensure_no_staging_repos_in_gopath() {
 }
 
 # Installs the specified go package at a particular commit.
+# Optionally specify an additional dependency constraint as "pkg@revision".
 kube::util::go_install_from_commit() {
   local -r pkg=$1
   local -r commit=$2
+  local -r extra_constraint=${3:-}
 
   kube::util::ensure-temp-dir
   mkdir -p "${KUBE_TEMP}/go/src"
@@ -565,6 +567,12 @@ kube::util::go_install_from_commit() {
     git fetch # TODO(spiffxp): workaround
     git checkout -q "${commit}"
     GOPATH="${KUBE_TEMP}/go" go get -d "${pkg}" #TODO(spiffxp): workaround
+    if [[ "${extra_constraint}" =~ ^(.+)@(.+)$ ]]; then
+      (
+        cd "${KUBE_TEMP}/go/src/${BASH_REMATCH[1]}"
+        git checkout -q "${BASH_REMATCH[2]}"
+      )
+    fi
     GOPATH="${KUBE_TEMP}/go" go install "${pkg}"
   )
   PATH="${KUBE_TEMP}/go/bin:${PATH}"

--- a/hack/update-bazel.sh
+++ b/hack/update-bazel.sh
@@ -25,8 +25,13 @@ source "${KUBE_ROOT}/hack/lib/init.sh"
 rm -f "${KUBE_ROOT}/pkg/generated/openapi/zz_generated.openapi.go"
 
 # The git commit sha1s here should match the values in $KUBE_ROOT/WORKSPACE.
-kube::util::go_install_from_commit github.com/kubernetes/repo-infra/kazel 4eaf9e671bbb549fb4ec292cf251f921d7ef80ac
-kube::util::go_install_from_commit github.com/bazelbuild/rules_go/go/tools/gazelle/gazelle 82483596ec203eb9c1849937636f4cbed83733eb
+kube::util::go_install_from_commit \
+    github.com/kubernetes/repo-infra/kazel \
+    4eaf9e671bbb549fb4ec292cf251f921d7ef80ac
+kube::util::go_install_from_commit \
+    github.com/bazelbuild/rules_go/go/tools/gazelle/gazelle \
+    82483596ec203eb9c1849937636f4cbed83733eb \
+    github.com/bazelbuild/buildtools@9455a9fa1081d94b2b3a11c66b7bde2ae5af3b86
 
 touch "${KUBE_ROOT}/vendor/BUILD"
 

--- a/hack/verify-bazel.sh
+++ b/hack/verify-bazel.sh
@@ -32,8 +32,13 @@ fi
 rm -f "${KUBE_ROOT}/pkg/generated/openapi/zz_generated.openapi.go"
 
 # The git commit sha1s here should match the values in $KUBE_ROOT/WORKSPACE.
-kube::util::go_install_from_commit github.com/kubernetes/repo-infra/kazel 4eaf9e671bbb549fb4ec292cf251f921d7ef80ac
-kube::util::go_install_from_commit github.com/bazelbuild/rules_go/go/tools/gazelle/gazelle 82483596ec203eb9c1849937636f4cbed83733eb
+kube::util::go_install_from_commit \
+    github.com/kubernetes/repo-infra/kazel \
+    4eaf9e671bbb549fb4ec292cf251f921d7ef80ac
+kube::util::go_install_from_commit \
+    github.com/bazelbuild/rules_go/go/tools/gazelle/gazelle \
+    82483596ec203eb9c1849937636f4cbed83733eb \
+    github.com/bazelbuild/buildtools@9455a9fa1081d94b2b3a11c66b7bde2ae5af3b86
 
 gazelle_diff=$(gazelle fix -build_file_name=BUILD,BUILD.bazel -external=vendored -mode=diff -repo_root="$(kube::realpath ${KUBE_ROOT})")
 kazel_diff=$(kazel -dry-run -print-diff -root="$(kube::realpath ${KUBE_ROOT})")

